packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -19,7 +19,7 @@ const String defaultMainPath = 'lib/main.dart';
 const String defaultAssetBasePath = '.';
 const String defaultManifestPath = 'pubspec.yaml';
 String get defaultDepfilePath => fs.path.join(getBuildDirectory(), 'snapshot_blob.bin.d');
-String get defaultApplicationKernelPath => fs.path.join(getBuildDirectory(), 'app.dill');
```

Comment:
Can't you just add 'track.dill' just before it is used if needed?

Reply:
ok let me try to do that again.
I am more confident with this version that we don't add the file extension multiple times and we never miss adding it because an explicit step is required to go from the path without and extension to a final path with an extension.